### PR TITLE
feat(tests): Add a linter to lint-staged (#165)

### PR DESCRIPTION
**Pull request by km4**
We can try manual linting all ts,tsx,js files  e.g. `npm run lint:js | lint:js:fix` on this time we get `✖ 43 problems (43 errors, 0 warnings)`

The linter should check only edited files in pre-commit.

https://github.com/xFAANG/askql/issues/165

### CODE REVIEW

**Comment by mhagmajer** [comment]
lgtm, great work @km4!

File: package.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
     "postbuild": "ts-node scripts/postBuild.ts",
     "prettier": "prettier --write --config prettier.config.js .",
     "prettier:check": "prettier --config prettier.config.js --check .",
-    "prettier:scripts": "prettier --write ./scripts"
+    "prettier:scripts": "prettier --write ./scripts",
+    "lint:js": "eslint . --ext .js,.tsx,.ts",
+    "lint:js:fix": "npm run lint --fix"
```
Comment (czerwinskilukasz1) [inline package.json:22]: I think it should be `lint:js` not `lint`

**Comment by czerwinskilukasz1** [comment]
@km4 , let's discuss your change. 
Is any rule apart from `@typescript-eslint/no-unused-vars` on?  It's very good if it's the only one, this is what keeps the PR small and make a discussion on rules to use separate from this one.

Can you relax the unused rule a bit more and disable an error for function arguments like this one?
image

(We can always switch it back, but I'm against it so let's disable it for now and then open a separate discussion)

File: package.json (hunk 1 above)
Comment (km4) [reply package.json]: Fix in 0243ea32fcf647c05662d9fd47117bf66d871c21

**Comment by km4** [comment]
> @km4 , let's discuss your change.
> Is any rule apart from `@typescript-eslint/no-unused-vars` on? It's very good if it's the only one, this is what keeps the PR small and make a discussion on rules to use separate from this one.
> 
> Can you relax the unused rule a bit more and disable an error for function arguments like this one?
> image
> 
> (We can always switch it back, but I'm against it so let's disable it for now and then open a separate discussion)

I can change error on warn for example and use some other args if needed or disable all no unused warn on the global rules.
image

Docs for args: https://eslint.org/docs/rules/no-unused-vars#args

**Comment by km4** [comment]
> Is eslint good with askjson and askcode files?

now eslint extensions only js,.tsx,.ts

**Comment by czerwinskilukasz1** [comment]
> > Is eslint good with askjson and askcode files?
> 
> now eslint extensions only js,.tsx,.ts

Just for now, just for now, we will get there :D

File: .eslintrc.js [hunk 2]
```diff
@@ -18,6 +18,6 @@ module.exports = {
     'no-useless-catch': 'off',
     'no-useless-escape': 'off',
     'valid-typeof': 'off',
-    '@typescript-eslint/no-unused-vars': 'error',
+    '@typescript-eslint/no-unused-vars': [1, { args: 'after-used' }],
```
Comment (czerwinskilukasz1) [inline .eslintrc.js:21]: I'd prefer `args: none` for now.
Also, please use 'warning' instead of a magic constant.

**Comment by czerwinskilukasz1** [comment]
I'd keep these without warnings for now:
image
but couldn't find a flag for it and the best workaround I found is to add:
`{ "argsIgnorePattern": "^_" }` (will not raise unused warning for variables starting with _)

File: .eslintrc.js (hunk 2 above)
Comment (km4) [reply .eslintrc.js]: Change on 462bf9f

**Comment by km4** [comment]
@czerwinskilukasz1 
> I'd get rid of these warnings for now:
> image

> but couldn't find a flag for it and the best workaround I found is to add:
> `{ "argsIgnorePattern": "^_" }` (will not raise unused warning for variables starting with _)

We can off all the no unused vars at this moment. Just change `1` to `0` or pass `'off'`

**Comment by czerwinskilukasz1** [comment]
Thank you @km4!
Let's add additional tweaks in separate PRs.

**Pull request merged**